src/EnergyPlus/MixedAir.cc

```diff
@@ -3660,42 +3630,45 @@ void VentilationMechanicalProps::CalcMechVentController(
     Real64 ZoneMinCO2 = 0.0;        // Minimum CO2 concentration in zone
     Real64 ZoneOAMin = 0.0;         // Minimum Zone OA flow rate when the zone is unoccupied (i.e. ZoneOAPeople = 0)
     Real64 ZoneOAMax = 0.0;         // Maximum Zone OA flow rate (ZoneOAPeople + ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::PerArea)])
-    Real64 CO2PeopleGeneration = 0; // CO2 generation from people at design level
-    MechVentOAMassFlow = 0.0;
+    Real64 MechVentOAMassFlow = 0.0;
 
     // Apply mechanical ventilation only when it is available/allowed
     if (GetCurrentScheduleValue(state, this->SchPtr) > 0) {
         Real64 SysOAMassFlow = 0.0; // System supply OA mass flow rate [kg/s]
         if (this->SystemOAMethod == DataSizing::SysOAMethod::IAQP) {
             // IAQP for CO2 control
             for (int ZoneIndex = 1; ZoneIndex <= this->NumofVentMechZones; ++ZoneIndex) {
-                int ZoneNum = this->VentMechZone(ZoneIndex);
+                auto &thisMechVentZone = this->VentMechZone(ZoneIndex);
+                int ZoneNum = thisMechVentZone.zoneNum;
                 SysOAMassFlow += state.dataContaminantBalance->ZoneSysContDemand(ZoneNum).OutputRequiredToCO2SP *
-                                 GetCurrentScheduleValue(state, this->ZoneOASchPtr(ZoneIndex));
+                                 GetCurrentScheduleValue(state, thisMechVentZone.ZoneOASchPtr);
             }
             MechVentOAMassFlow = SysOAMassFlow;
```

Comment:
Might? be able to substitute MechVentOAMassFlow for SysOAMassFlow but not sure at this point. Leave for now.